src/toDoList.js

```diff
@@ -2,39 +2,153 @@ import React from 'react';
+      this.doRenderSearchBar =  this.doRenderSearchBar.bind(this);
+  }
+
+  editItem(id) {
+    this.setState({
+      isUpdate : true,
+      id : id
+    });
   }
 
   addItem() {
-    let inputText = this.state.value;
-    this.state.items = [...this.state.items, inputText];
-    this.state.value = '';
+    let inputText = this.state.value; //object destructring
+    let newItems = this.state.items;
+    let searchItems = this.state.searchResults;
+    let isCheck = false;
+
+    if(this.state.isUpdate) {
+      newItems = newItems.map((value, i) => {
+        if(value === searchItems[this.state.id] && !isCheck) {
+          isCheck = true;
+          return inputText;
+        }
+        return value;
+      });
+      searchItems[this.state.id] = inputText;
+    } else {
+      newItems = [...this.state.items, inputText]
+      searchItems = [...newItems]
+    }
+    //this.state.value = '';  dont change state variables directly... use setState instead
     this.setState({
       value : '',
-      items : this.state.items
+      items : newItems,
+      isUpdate : false,
+      id : 9999,
+      searchResults: searchItems
     });
   }
 
-  displayList(){
-    return (this.state.items.slice(0).reverse().map((item,i) => <li key={i}>{item}</li>));
+  removeItem(id) {
+    let isCheck = false;
+    let newItems = this.state.items.filter((item, i) => {
+      if(item === this.state.searchResults[id] && !isCheck) {
+        isCheck = true;
+      } else {
+        return item;
+      }
+    });
+    let searchItems = this.state.searchResults.filter((item, i) => i !== id);
+    this.setState({
+      items : newItems,
+      searchResults: searchItems
+    });
+  }
+
+  displayList() {
+    return (<ul>{this.state.searchResults.map((item, i) => {
+      return (
+        <li key={i}>{item}
+        <button
+          onClick={this.removeItem.bind(this, i)}
+          type="button"
+          className="btn btn-primary">Remove
+        </button>
+        <button
+          onClick={this.editItem.bind(this, i)}
+          type="button"
+          className="btn btn-primary">Edit
+        </button>
+      </li>);
+    })}</ul>);
+  }
+
+  doRenderAddOrUpdateButton() {
+    let buttonText = "Add";
+    if(this.state.isUpdate){buttonText = "Edit";}
+    return (
+      <button
+        onClick={this.addItem}
+        type="button"
+        className="btn btn-primary">{buttonText}
+      </button>
+    );
   }
 
-  render() {
-    let items = this.state.items;
+  doRenderSearchBar() {
+    return (
+      <input
+      className="input"
+      size="40"
+      type="text"
+      placeholder="Search..."
+      onChange={this.onChangeFunctions('search')}
+      value={this.state.searchValue}
+    />
+    );
+  }
+
+  onChangeFunctions(action) {
+    return (
+      action === 'search' ?
+      e => {
+        let searchResults = null;
+        searchResults = this.state.items.filter((value, i) => value.includes(e.target.value));
```

Comment:
Get rid of includes. It's not supported in IE. Even when coding in JS, it should be cross browser.

Reply:
used indexOf instead